components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/AdminRolePermissionsUpdateListener.java

```diff
@@ -0,0 +1,191 @@
+            List<Permission> systemPermissions =  new ArrayList<>();
+            for (String scope : internalScopes) {
+                systemPermissions.add(new Permission(scope));
+            }
+            RoleManagementService roleManagementService = ApplicationManagementServiceComponentHolder.getInstance()
+                    .getRoleManagementServiceV2();
+            roleManagementService.updatePermissionListOfRole(adminRoleId, systemPermissions, new ArrayList<>(),
+                    tenantDomain);
+        } catch (IdentityRoleManagementException e) {
+            throw new IdentityApplicationManagementException("Error while update admin role permissions", e);
+        }
+    }
+
+    /**
+     * Add admin role to console app as associated role.
+     *
+     * @param adminRoleId Admin role id.
+     * @param serviceProvider Service provider.
+     * @param tenantDomain Tenant domain.
+     * @throws IdentityApplicationManagementException if an error occurs while adding admin role to console app.
+     */
+    private void addAdminRoleToConsoleAppAsAssociatedRole(String adminRoleId, ServiceProvider serviceProvider,
+                                                          String tenantDomain)
+            throws IdentityApplicationManagementException {
+
+        AssociatedRolesConfig associatedRolesConfig = new AssociatedRolesConfig();
+        associatedRolesConfig.setAllowedAudience(ORGANIZATION);
+        serviceProvider.setAssociatedRolesConfig(associatedRolesConfig);
+        ApplicationManagementServiceImpl.getInstance().addAssociatedRoleToApplication(serviceProvider, adminRoleId,
+                tenantDomain);
+    }
+
+    /**
+     * Get Admin role id.
+     *
+     * @param tenantDomain Tenant domain.
+     * @return Admin role id.
+     * @throws IdentityApplicationManagementException if an error occurs while retrieving admin role id.
+     */
+    private String getAdminRoleId(String tenantDomain) throws IdentityApplicationManagementException {
+
+        String orgId;
+        try {
+            orgId = ApplicationManagementServiceComponentHolder.getInstance().getOrganizationManager()
+                    .resolveOrganizationId(tenantDomain);
+        } catch (OrganizationManagementException e) {
+            throw new IdentityApplicationManagementException("Error while retrieving organization id from tenant " +
+                    "domain : " + tenantDomain, e);
+        }
+        if (orgId == null) {
+            throw new IdentityApplicationManagementException("Error while retrieving organization id from tenant " +
+                    "domain : " + tenantDomain);
+        }
+        RoleManagementService roleManagementService = ApplicationManagementServiceComponentHolder.getInstance()
+                .getRoleManagementServiceV2();
+        try {
+            return roleManagementService.getRoleIdByName("admin", ORGANIZATION, orgId, tenantDomain);
```

Comment:
"admin" role name should be taken from the config in the system.

AFAIR, the default admin role can be configured in the product's xml file

Check user-mgt.xml

```
    <Realm>
        <Configuration>
            <AddAdmin>true</AddAdmin>
            <AdminRole>admin</AdminRole>
            <AdminUser>
                <UserName>admin</UserName>
                <Password>admin</Password>
            </AdminUser>
            <EveryOneRoleName>everyone</EveryOneRoleName>
```
Value in AdminRole config should be used here

Reply:
fixed